Library/Homebrew/formula_installer.rb

```diff
@@ -212,6 +213,10 @@ def install
       opoo "#{formula.full_name}: #{old_flag} was deprecated; using #{new_flag} instead!"
     end
 
+    invalid_flag_names.each do |flag|
+      opoo "#{formula.full_name}: #{flag} is invalid for this formula and will be ignored!"
```

Comment:
ok I change to hard fail then - if we are sure users don't rely on invalid options silently being ignored behaviour? Oh, and do you want a `--no-invalid-option-warnings` flag to override?